spec/system/assignment/index_spec.rb

```diff
@@ -65,7 +65,7 @@ def bg_variable(var)
     context 'when clicking on an empty day' do
       let(:date) { Time.zone.today.change(day: 14) }
       let(:new_path) do
-        new_roster_assignment_path roster_id: roster.id,
+        new_roster_assignment_path roster_id: roster,
```

Comment:
Is this some feature of `friendly_id`? Shouldn't this be `roster: roster` or `roster_id: roster.to_param`?

Reply:
`to_param` gets called on everything passed into route helpers. So, without FriendlyId, these are equivalent.

But the key needs to match the route fragment name, e.g.

```
       new_roster_assignment GET    /rosters/:roster_id/assignments/new(.:format)   assignments#new
```

But, the reason it looks unfamiliar is probably because, by virtue of being a resource route, this also works:

 ```suggestion
         new_roster_assignment_path roster,
```

To sum up:

```ruby
r = Roster.first
app.new_roster_assignment_path(roster: r)
# /home/matt/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/actionpack-7.0.7.2/lib/action_dispatch/journey/formatter.rb:44:in `path':
# No route matches {:action=>"new", :controller=>"assignments", :roster=>#<Roster id: 1, name: "Transit IT", created_at: "2023-09-01 07:33:34.000000000 -0400", updated_at: "2023-09-06 10:16:25.000000000 -0400", fallback_user_id: 4, switchover: 1000, phone: nil, slug: "transit-it">},
# missing required keys: [:roster_id] (ActionController::UrlGenerationError)

app.new_roster_assignment_path(roster_id: r)
#=> "/rosters/transit-it/assignments/new"
app.new_roster_assignment_path(roster_id: r.to_param)
#=> "/rosters/transit-it/assignments/new"
app.new_roster_assignment_path(roster_id: r.id)
#=> "/rosters/1/assignments/new"
app.new_roster_assignment_path(roster_id: r.id.to_param)
#=> "/rosters/1/assignments/new"
app.new_roster_assignment_path(r)
#=> "/rosters/transit-it/assignments/new"
app.new_roster_assignment_path(r.to_param)
#=> "/rosters/transit-it/assignments/new"
app.new_roster_assignment_path(r.id)
#=> "/rosters/1/assignments/new"
```